### PR TITLE
openttd: update to 12.2.

### PR DESCRIPTION
**Pull request by Terracotta-Lite**
Testing the changes
- I tested the changes in this PR: **NO**

### CODE REVIEW

File: srcpkgs/openttd/template [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Template file for 'openttd'
 pkgname=openttd
-version=1.10.3
+version=12.2
 revision=4
 _gfxver=0.6.0
 _sfxver=0.2.3
```
Comment (classabbyamp) [inline srcpkgs/openttd/template:6]: there are new versions of [opengfx](https://www.openttd.org/downloads/opengfx-releases/latest.html) and [opensfx](https://www.openttd.org/downloads/opensfx-releases/latest) available, too

**Review by classabbyamp — COMMENTED** [review]
the checksums will need to be updated as well

**Review by Idesmi — COMMENTED** [review]
OpenTTD has since switched from gnu-configure to CMake.
Also it would be best to use SDL2 by default.
I have a commit ready for the update, prepared after seeing the post on Reddit (and before seeing this PR): https://github.com/Idesmi/void-packages/commit/63249b80f16575c70d301271f342002c70d39a59

**Comment by paper42** [comment]
@Idesmi please open your own PR that will close this one.

**Comment by Idesmi** [comment]
> @Idesmi please open your own PR that will close this one.

@Terracotta-Lite Would this be fine with you?

**Pull request closed without merging**